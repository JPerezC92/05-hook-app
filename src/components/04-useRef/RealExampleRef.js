import React, { useState } from "react";
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";
import "./useRef.css";

const RealExampleRef = () => {
  const [show, setShow] = useState(false);
  return (
    <div>
      <h1>RealExampleRef</h1>

      {show && <MultipleCustomHooks />}

      <button
        className="btn btn-primary mt-4"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>
    </div>
  );
};

export default RealExampleRef;
