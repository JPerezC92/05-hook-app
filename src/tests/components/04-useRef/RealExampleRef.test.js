import { shallow } from "enzyme";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe("Pruebas en el componente <RealExampleRef />", () => {
  const wrapper = shallow(<RealExampleRef />);

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("Debe de mostrar el elemento <MultipleCustomHooks />", () => {
    wrapper.find("button").at(0).simulate("click");

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
