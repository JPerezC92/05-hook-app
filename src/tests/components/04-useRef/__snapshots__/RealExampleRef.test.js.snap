// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <RealExampleRef /> Debe de mostrar el elemento <MultipleCustomHooks /> 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Pruebas en el componente <RealExampleRef /> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
